src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -94,4 +95,107 @@ public void testOffsetAtRead(String cigar, int expectedReferencePos, int posInRe
             SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
             Assert.assertEquals(sam.getReferencePositionAtReadPosition(posInRead), expectedReferencePos);
     }
+
+    @Test(dataProvider = "referenceAtReadData")
+    public void testDeepCopyRef(String cigar, int expectedReferencePos, int posInRead) {
+
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        testDeepCopy(sam);
+    }
+
+    @Test(dataProvider = "offsetAtReferenceData")
+    public void testDeepCopyOffset(String cigar, int posInReference, int expectedPosInRead, boolean returnLastBaseIfDeleted) {
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        testDeepCopy(sam);
+    }
+
+    @Test(dataProvider = "serializationTestData")
+    public void testDeepCopySerialization( final File inputFile ) throws Exception {
```

Comment:
This test case seems mis-namedm as it's not testing serialization -- shouldn't it just be called testDeepCopyAttributes? (rather than calling a method of that name).